plugins/gemalto.c

```diff
@@ -168,17 +166,25 @@ struct gemalto_data {
 
 	struct ofono_netreg *netreg;
 
-	void *device; /* struct mbim_device* or struct qmi_device* */
+	/* struct mbim_device* or struct qmi_device* */
+	union {
+	  struct mbim_device *mbim;
+	  struct qmi_device  *qmi;
+	} device;
```

Comment:
please lose the struct, use instead just the two variables.
Now that you took the time to replace device with device->mbim_device and device->qmi_device as adequate, it should be easy to do a massive replace in the entire file.